anonymizer/OpenDiffix.Service/AggregationHooks.fs

```diff
@@ -189,20 +190,23 @@ module StarBucket =
           |> Value.unwrapBoolean
 
         if not isAlreadyMerged && isLowCount lowCountIndex bucket then
+          bucketCount <- bucketCount + 1
           bucket |> mergeAllAggregatorsInto starBucket
       )
 
     let executionContext = starBucket.ExecutionContext
 
-    // NOTE this is also true when there is no star bucket (no suppression)
-    //      this is because of how the initial value (`null`) of the
-    //      `DiffixLowCount` aggregator is translated to `Boolean true`
     let isStarBucketLowCount =
       starBucket.Aggregators.[lowCountIndex].Final(executionContext)
       |> Value.unwrapBoolean
```

Comment:
Does `isStarBucketLowCount` depend on `bucketCount`? There's a chance that a single suppressed bucket can become high count due to the different seed.

Reply:
I see that only anon count is exported. Ok in that case.

Reply:
it doesn't.

`bucketCount` is "count of buckets in star bucket", not related to `low count filtering` or any other notion of "count" we have.

I tried to find a nicer name for this var, but couldn't so I kept it simple.

> There's a chance that a single suppressed bucket can become high count due to the different seed.

this is exactly why we're introducing the isStarBucketSingleBucket condition. If it's `true` it prevents such an "un-suppressed" bucket to surface.

Reply:
How about simplifying to:

```
    let isStarBucketLowCount =
      (if bucketCount < 2 then
        Boolean true
      else
        starBucket.Aggregators.[lowCountIndex].Final(executionContext))
      |> Value.unwrapBoolean
```

and removing the extra flag?

Reply:
We derive state closer to the source.

Reply:
I would disagree. `bucketCount < 2` isn't exactly `star bucket low count`, it's a slightly different condition.

I'm actually thinking (I've already done it, now I'm thinking if it's correct), on rewriting it to a sequence of steps like:
1/ filter `buckets` seq, to only have those which satisfy the criteria (low count & not merged)
2/ make a second pass through the filtered seq to actually perform the mergeing.

Then the filtered seq becomes a very natural source of truth for the needed condition

Reply:
PTAL, when you're feeling better https://github.com/diffix/desktop/pull/295/commits/64692ebcaf21b9b2e0f845e0b1be9a2691587e94

Reply:
> I would disagree. bucketCount < 2 isn't exactly star bucket low count, it's a slightly different condition.

Variable can be renamed to something else, such as `starBucketExists`.

With multiple iteration we replay the sequence which is something I'd avoid.